crl/updater/updater.go

```diff
@@ -217,25 +281,41 @@ func (cu *crlUpdater) updateShard(ctx context.Context, atTime time.Time, issuerN
 			RevokedBefore: timestamppb.New(atTime),
 		})
 		if err != nil {
-			return fmt.Errorf("connecting to SA: %w", err)
+			return fmt.Errorf("GetRevokedCerts: %w", err)
 		}
 
-		for {
-			entry, err := saStream.Recv()
-			if err != nil {
-				if err == io.EOF {
-					break
-				}
-				return fmt.Errorf("retrieving entry from SA: %w", err)
-			}
-			crlEntries = append(crlEntries, entry)
+		n, err := addFromStream(crlEntries, saStream)
+		if err != nil {
+			return fmt.Errorf("streaming GetRevokedCerts: %w", err)
 		}
 
 		cu.log.Infof(
 			"Queried SA for CRL shard: id=[%s] expiresAfter=[%s] expiresBefore=[%s] numEntries=[%d]",
-			crlID, chunk.start, chunk.end, len(crlEntries))
+			crlID, chunk.start, chunk.end, n)
```

Comment:
Using `n` returned by `addFromStream()`, instead of `len(crlEntries)` is a pretty subtle change of behavior. One that I want to make sure was intentional given the `crlEntries` is deduplicated where as `n` is explicitly not:

> ...Returns the number of entries received from the stream, regardless of duplicate status.

Similar on line `316`.

Reply:
Yep, thanks for pointing this out! My thought process was that since we're logging each chunk separately, plus the explicit shard query separately, it's more informative to log the number of results from those queries individually. The total number of entries (after deduping) gets logged by the CA at signing time.